app/src/test/java/com/mstudio/superheroarch/repository/RickAndMortyRepositoryTest.kt

```diff
@@ -60,4 +60,11 @@ class RickAndMortyRepositoryTest {
         `when`(rickAndMortyApiMock.getSingleEpisode(1)).thenThrow(Exception())
         rickAndMortyRepository.getSingleEpisode(1)
     }
+
+    @Test
+    fun `given rickAndMorty repository, when set character as favorite, then update user as favorite in database`() = runTest {
```

Comment:
this test is useless, you are testing that mock is working, "when update is called unit is returned, configure mock to return unit when update is called, then I call the mocked method and mock is working returning Unit, which is the thing I just configured"
you are not checking the database as the title suggest. 

To implement this test you'll need something more complex, you'll need a mock database or a method that mocks the db, then you'll need to call the setFavorite method and then check that if you get that character, when it comes from db, is marked as favorite